File: linkedList.cpp
#include <iostream>

using namespace std;

struct Node {
    int value;
    struct Node* next;
    struct Node* prev;
};

class LinkedList {
   private:
    struct Node* start;
    struct Node* end;

   public:
    LinkedList(int val) {
        struct Node node;

        node.value = val;
        node.next = NULL;

        start = &node;
        end = &node;

        cout << "constructor start = " << start->value << "&start, &end" << &start << &end << "\n";
    }

    void insertNode(int value) {
        cout << "inserting " << value << " start = " << start->value << endl;

        struct Node newNode;

        newNode.value = value;

        end->next = &newNode;
        end = end->next;
    }

    int iterate() {
        cout << start->value;
        return 1;

        Node* ptr = start;

        while (ptr->next != NULL) {
            cout << "value " << ptr->value << " ";
            ptr = ptr->next;
        }
    }
};

int main() {
    LinkedList ll(6);

    ll.insertNode(7);
    ll.insertNode(8);
    ll.insertNode(9);
    ll.insertNode(10);
    ll.insertNode(11);
    ll.insertNode(12);

    ll.iterate();
}

File: LinkedLists/LinkedList.py
from typing import Union


class Node:
    def __init__(self, value: int) -> None:
        self.next: Union[None, Node] = None
        self.value: int = value


class LinkedList:
    def __init__(self) -> None:
        self.start = None

    def insert(self, value: int):
        node = Node(value)

        if not self.start:
            self.start = node
            return self

        ptr = self.start

        while ptr.next is not None:
            ptr = ptr.next

        ptr.next = node

        return self

    def traverse(self):
        string = ""

        ptr = self.start

        while ptr is not None:
            string += f"{ptr.value} -> "
            ptr = ptr.next

        # print(string[:-4])

        return string[:-4]


if __name__ == "__main__":
    ll = LinkedList()

    for i in range(1, 11):
        ll.insert(i)

    ll.traverse()
